modules/navier_stokes/src/fvkernels/INSFVMomentumAdvection.C

```diff
@@ -127,12 +152,17 @@ INSFVMomentumAdvection::computeResidualsAndAData(const FaceInfo & fi)
       // neighbor side:
       // rho * v_superficial / eps_neigh * v_superficial = rho * v_interstitial_neigh *
       // v_superficial
-      const auto elem_coeff = _normal * v_face * rho_face / eps_elem_face;
-      const auto neighbor_coeff = _normal * v_face * rho_face / eps_neighbor_face;
+      const auto elem_coeff = vdotn * rho_face / eps_elem_face;
+      const auto neighbor_coeff = vdotn * rho_face / eps_neighbor_face;
       _ae = elem_coeff * d_var_elem_face_d_elem_dof;
       _elem_residual = elem_coeff * var_elem_face;
       _an = -neighbor_coeff * d_var_neighbor_face_d_neighbor_dof;
       _neighbor_residual = -neighbor_coeff * var_neighbor_face;
+      if (_approximate_as)
+      {
+        _ae = _cs / fi.elem().n_sides() * rho_face / eps_elem_face;
```

Comment:
what is the logic behind dividing by the number of element sides?
if we have tris and tets in the same mesh, _cs is effectively larger in the tri region?

same for 2D / 3D

Reply:
nevermind you sum over sides

Reply:
You should think about the sum over all the faces. For an internal element, the result should be that the elemental contribution is c * rho / eps, e.g. sum(c * rho / eps / n-sides) -> c * rho / eps

Reply:
You surmised the reason even before I responded!